extensions/get_replicate_average.py

```diff
@@ -0,0 +1,178 @@
+        print(f"Error: file not found ({t})")
+        return 1
+
+    output_fh.write("Sequence name\t")
+
+    # Get the first line's base sequence
+    current_line = names_fh.readline()
+    current_line_list = create_list(current_line, "")
+    base_sequence = current_line_list[0]
+
+    # Write the first base sequence
+    base_sequences.append(base_sequence)
+    output_fh.write(f"{base_sequence}")
+    sequence_dict[base_sequence] = []
+
+    # Get the rest of the base sequences (if there are any)
+    while True:
+        # Get the current line
+        current_line = names_fh.readline()
+        if current_line == "":
+            break
+
+        # Get the base sequence on the current line
+        current_line_list = create_list(current_line, "")
+        base_sequence = current_line_list[0]
+
+        # Use base sequence for the following:
+        # 1. Append it to the list of base sequences
+        # 2. Write it to the output file
+        # 3. Add it as a key in the sequence dictionary
+        base_sequences.append(base_sequence)
+        output_fh.write("\t")
+        output_fh.write(f"{base_sequence}")
+        sequence_dict[base_sequence] = []
+
+    output_fh.write("\n")
+
+    # Get list of sequence names from input score file
+    sequence_names_list = create_list(score_fh.readline(), "Sequence name")
+
+    # Loop through score file & record the replicate averages
+    while True:
+        base_sequence_found = False
+
+        # Check if at EOF
+        current_line = score_fh.readline()
+        if current_line == "":
+            break
+
+        # Create list out of values on current row
+        current_line_list = create_list(current_line, "")
+
+        # Get current peptide and remove from list
+        current_peptide = current_line_list[0]
+        current_line_list.pop(0)
+
+        # Loop through sequence names
+        sequence_names_index = 0
+        while sequence_names_index < len(sequence_names_list):
+            # Loop through base sequences
+            base_sequence_index = 0
+            while base_sequence_index < len(base_sequences):
+                # Check if sequence name contains base sequence name
+                if sequence_names_list[sequence_names_index].find(base_sequences[base_sequence_index]) != -1:
+                    base_sequence_found = True
+
+                    # Create/update sequence dictionary entry with:
+                    # Key: sequence name
+                    # Entry: list of replicate values
+                    sequence_dict[base_sequences[base_sequence_index]].append\
+                        (current_line_list[sequence_names_index])
+
+                base_sequence_index += 1
+
+            # Check if the base sequence was not found after the search
+            if not base_sequence_found:
+                print("Error: base sequence not found")
+
+                names_fh.close()
+                score_fh.close()
+                output_fh.close()
```

Comment:
If you open files using "with" then you don't have to worry about explicitly closing files like this. For example:

with open("names_file", "r") as names_fh: